### PR TITLE
feat: implementation of ses configuration set delete

### LINKED ISSUE #9773: Support required for ses-configuration-set Delete Action
### Describe the feature

Need support for Delete Action for ses-configuration-set

### Extra information or context

https://docs.aws.amazon.com/ses/latest/APIReference/API_DeleteConfigurationSet.html

### PR DESCRIPTION
**Pull request by archinksagar**
Implementation of Delete action for ses-configuration-set
Closes #9773

### CODE REVIEW

File: c7n/resources/ses.py [hunk 1]
```diff
@@ -87,6 +87,34 @@ def process(self, resources):
...
+    schema = type_schema('delete')
+    permissions = ("ses:DeleteConfigurationSet",)
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('ses')
+        for resource in resources:
+            self.manager.retry(
+                client.delete_configuration_set,
+                ConfigurationSetName=resource['Name'],
+                ignore_err_codes=("CannotDeleteException",)
```
Comment (jerryhxu) [inline c7n/resources/ses.py:114]: Based on boto3 doc: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ses/client/delete_configuration_set.html, there is not "CannotDeleteException", only "ConfigurationSetDoesNotExistException".

**Review by jerryhxu — CHANGES_REQUESTED** [review]

File: c7n/resources/ses.py (hunk 1 above)
Comment (archinksagar) [reply c7n/resources/ses.py]: @jerryhxu changed it to correct exception. thank you.

File: tests/test_ses.py [hunk 2]
```diff
@@ -43,6 +43,27 @@ def test_ses_configuration_set_delivery_options(self):
...
+            {
+                "name": "ses-configuration_set-delete-test",
+                "resource": "ses-configuration-set",
+                "actions": [{"type": "delete"}],
+            }, session_factory=session_factory
+        )
+
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
+        p = self.load_policy(
```
Comment (PratMis) [inline tests/test_ses.py:58]: It would be better to assert on the resource name and check whether that resource exists or not?

**Review by kapilt — APPROVED** [review]
lgtm, thanks!

**Pull request merged**